crates/pathfinder/src/storage/state.rs

```diff
@@ -51,28 +57,103 @@ impl L1StateTable {
                         :ethereum_transaction_index,
                         :ethereum_log_index
                     )",
-                named_params! {
-                    ":starknet_block_number": update.block_number.0,
-                    ":starknet_global_root": update.global_root.0.as_be_bytes(),
-                    ":ethereum_block_hash": &update.origin.block.hash.0[..],
-                    ":ethereum_block_number": update.origin.block.number.0,
-                    ":ethereum_transaction_hash": &update.origin.transaction.hash.0[..],
-                    ":ethereum_transaction_index": update.origin.transaction.index.0,
-                    ":ethereum_log_index": update.origin.log_index.0,
-                },
-            )
-            .context("Insert L1 state update")?;
+            named_params! {
+                ":starknet_block_number": update.block_number.0,
+                ":starknet_global_root": update.global_root.0.as_be_bytes(),
+                ":ethereum_block_hash": &update.origin.block.hash.0[..],
+                ":ethereum_block_number": update.origin.block.number.0,
+                ":ethereum_transaction_hash": &update.origin.transaction.hash.0[..],
+                ":ethereum_transaction_index": update.origin.transaction.index.0,
+                ":ethereum_log_index": update.origin.log_index.0,
+            },
+        );
+
+        match result {
+            Ok(_) => Ok(()),
+            Err(err) => {
+                if err.to_string().contains("A PRIMARY KEY constraint failed") {
+                    let latest = L1_LATEST_REORG
+                        .lock()
+                        .map_err(|_| anyhow::anyhow!("L1 accounting lock is poisoned"))
```

Comment:
```suggestion
                        .unwrap_or_else(|e| e.into_inner())
```

Since the value inside is just a tuple ... I don't yet understand what it is but regardless.